### PR TITLE
[Incremental] Honor -driver-always-rebuild-dependents

### PR DESCRIPTION
**Pull request by davidungar**
In order to pass legacy tests, honor -driver-always-rebuild-dependents

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please test

**Review by artemcm — APPROVED** [review]

File: Sources/SwiftDriver/Incremental Compilation/IncrementalCompilationState.swift [hunk 1]
```diff
@@ -330,26 +333,32 @@ extension IncrementalCompilationState {
...
     reportIncrementalDecision: ((String, TypedVirtualPath?) -> Void)?
     ) -> Set<TypedVirtualPath> {
     // Collect the files that will be compiled whose dependents should be schedule
     let cascadingFiles: [TypedVirtualPath] = changedInputs.compactMap { input, status in
       let basename = input.file.basename
-      switch status {
-      case .needsCascadingBuild:
+      switch (status, alwaysRebuildDependents) {
+
+       case (_, true):
```
Comment (artemcm) [inline Sources/SwiftDriver/Incremental Compilation/IncrementalCompilationState.swift:344]: Can we add a test that exercises the `(.upToDate, true)` to ensure this is not regressed in the future?

Comment (davidungar) [reply Sources/SwiftDriver/Incremental Compilation/IncrementalCompilationState.swift:344]: One of the legacy lit tests already does so; that's how I found the issue. But, let me see if I can add an XC test.

Comment (davidungar) [reply Sources/SwiftDriver/Incremental Compilation/IncrementalCompilationState.swift:344]: And thanks!

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by artemcm** [comment]
Thank you, @davidungar!

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
Fixes rdar://72157018

**Pull request merged**